main.rb

```diff
@@ -0,0 +1,159 @@
+# rubocop:disable Metrics/CyclomaticComplexity
+# rubocop:disable Metrics/PerceivedComplexity
+module Enumerable
+  def my_each
+    return to_enum unless block_given?
+
+    i = 0
+    arr = to_a
+    while i <= arr.length - 1
+      yield(arr[i])
+      i += 1
+    end
```

Comment:
- [ ] when a block is given when self is an array returns the array itself after calling the given block once for each element in self

- [ ]  when a block is given when self is a range returns the range itself after calling the given block once for each element in self